packages/langium/src/parser/cst-node-builder.ts

```diff
@@ -225,7 +225,7 @@ export class CompositeCstNodeImpl extends AbstractCstNode implements CompositeCs
 
     private get firstNonHiddenNode(): CstNode | undefined {
         for (const child of this.content) {
-            if (!child.hidden) {
+            if (!child.hidden && !isNaN(child.offset)) {
```

Comment:
Would it be possible to draft a test grammar for this edge case you are mentioned in the PR description?